packages/react/src/SubNav/SubNav.tsx

```diff
@@ -176,56 +177,94 @@ type SubNavLinkProps = {
 } & PropsWithChildren<React.HTMLProps<HTMLAnchorElement>> &
   BaseProps<HTMLAnchorElement>
 
-const SubNavLink = ({
+const SubNavLinkWithSubmenu = ({
   children,
   href,
   'aria-current': ariaCurrent,
   'data-testid': testId,
   className,
   ...props
 }: SubNavLinkProps) => {
-  const hasSubMenu = Children.toArray(children).some(child => {
+  const [isExpanded, setIsExpanded] = useState(false)
+  const onFocusChange = useCallback((isFocused: boolean) => {
+    if (!isFocused) {
+      setIsExpanded(false)
+    }
+  }, [])
+  const submenuId = useId()
+  const ref = useIsChildFocused<HTMLDivElement>(onFocusChange)
```

Comment:
Related to https://github.com/primer/brand/pull/679/files#r1706597663

This is probably another reason to remove the hook, it doesn't behave like an actual hook where given the required argument needing to be declared first, it can't be called at the beginning of the closure. Surprised that didn't trigger the rules of hook linter warning 🤔

Reply:
Could you expand on this a bit please, I'm not sure I understand why this isn't behaving like a hook?

My understanding is that hooks are allowed to consume the results of other hooks (we do it all the time), and they don't need to be called at the beginning of the containing component, they just [need to be called at the top level](https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level).

Happy to have a call if that's easier 🙂